app/controllers/api/v1/attachments_controller.rb

```diff
@@ -39,15 +48,12 @@ def create
       #{json_schema(Api::V1::AttachmentRepresenter, include: :readable)}
     EOS
     def destroy
-      attachment = @exercise.attachments.find_by! asset: params[:filename]
+      exercise = Exercise.visible_for(user: current_api_user).with_id(params[:exercise_id]).first!
+      attachment = exercise.attachments.find_by! asset: params[:filename]
       standard_destroy(attachment)
     end
 
     protected
 
-    def get_exercise
```

Comment:
Seems to me that this was just moved up into the methods so it could probably have been kept.

Reply:
not quite, `delete` uses the bang find so that it'll 404.  `create` adds a new version if it's not found